### PR TITLE
Mo 74 in resource reservation add repeatable

### PR DESCRIPTION
**Pull request by AndriiK139MCAF**
Don't know what to do with Flake8 and Pylint errors

### CODE REVIEW

File: resource_reservation/models/resource_reservation.py [hunk 1]
```diff
@@ -1,5 +1,7 @@
 """Required import for the functionality"""
 from odoo import models, fields, api, exceptions, _
+from datetime import timedelta
```
Comment (MORPSmcaf) [inline resource_reservation/models/resource_reservation.py:3]: place line 3 before line 2

File: resource_reservation/models/resource_reservation.py [hunk 2]
```diff
@@ -176,11 +192,40 @@ def check_future_start_date(self):
...
+        for record in self:
+            if record.repeat_until:
+                limit_date = (fields.Datetime.now() +
+                              timedelta(days=90))
+                if record.repeat_until > limit_date:
+                    raise ValidationError(
+                        "You cannot select repeat "
+                        "date more than 3 months ahead.")
+                if record.repeat_until <= record.end_datetime:
+                    raise ValidationError(
```
Comment (MORPSmcaf) [inline resource_reservation/models/resource_reservation.py:206]: add        (_         to ValidationError and  in the end      ") 
like
raise exceptions.ValidationError(_("You cannot select date "
                        "before End Date & Time."))

File: resource_reservation/models/resource_reservation.py [hunk 3]
```diff
@@ -142,22 +152,28 @@ def update_booking_status_confirm(self):
...
+        creating multiple Resource Reservation
+        '''
+        new_record = super(ResourceReservation, self).create(vals)
+        if vals.get('repeat') == 'weakly':
+            new_vals = vals.copy()
+            repit_date = fields.Datetime.from_string(new_vals['repeat_until'])
+            repit_date -= timedelta(days=7)
+            repit_date_str = fields.Datetime.to_string(repit_date)
+            while new_vals.get('end_datetime') <= repit_date_str:
+                dont_usk_brake = 0
```
[the remark this answers is not in the document]
Comment (AndriiK139MCAF) [reply resource_reservation/models/resource_reservation.py:166]: and fix code what it shood doing

File: resource_reservation/models/resource_reservation.py (hunk 2 above)
Comment (AndriiK139MCAF) [reply resource_reservation/models/resource_reservation.py:206]: and delete  line 3

File: resource_reservation/models/resource_reservation.py [hunk 4]
```diff
@@ -176,11 +192,40 @@ def check_future_start_date(self):
...
                                                    "not allowed."))
 
+    @api.constrains('repeat_until')
+    def _check_repeat_until(self):
+        for record in self:
+            if record.repeat_until:
+                limit_date = (fields.Datetime.now() +
+                              timedelta(days=90))
+                if record.repeat_until > limit_date:
+                    raise ValidationError(
```
[the remark this answers is not in the document]
Comment (AndriiK139MCAF) [reply resource_reservation/models/resource_reservation.py:202]: and delete line 3

**Comment by AndriiK139MCAF** [comment]
Fix error  when you try create repeat Reservation without filling "Repeat until"

File: resource_reservation/models/resource_reservation.py [hunk 5]
```diff
@@ -176,11 +192,40 @@ def check_future_start_date(self):
...
+        for record in self:
+            if record.repeat_until:
+                limit_date = (fields.Datetime.now() +
+                              timedelta(days=90))
+                if record.repeat_until > limit_date:
+                    raise exceptions.ValidationError(
+                        "You cannot select repeat "
+                        "date more than 3 months ahead.")
+                if record.repeat_until <= record.end_datetime:
+                    raise exceptions.ValidationError(
```
Comment (MORPSmcaf) [inline resource_reservation/models/resource_reservation.py:206]: add (_ symbols after ValidationError(
and )
after ...ate & Time.")

File: resource_reservation/models/resource_reservation.py [hunk 6]
```diff
@@ -176,11 +192,40 @@ def check_future_start_date(self):
...
                                                    "not allowed."))
 
+    @api.constrains('repeat_until')
+    def _check_repeat_until(self):
+        for record in self:
+            if record.repeat_until:
+                limit_date = (fields.Datetime.now() +
+                              timedelta(days=90))
+                if record.repeat_until > limit_date:
+                    raise exceptions.ValidationError(
```
Comment (MORPSmcaf) [inline resource_reservation/models/resource_reservation.py:202]: add (_ symbols after ValidationError(
and )
after ....onths ahead.")

File: resource_reservation/models/resource_reservation.py (hunk 2 above)
Comment (AndriiK139MCAF) [reply resource_reservation/models/resource_reservation.py]: added ValidationError(_())

File: resource_reservation/models/resource_reservation.py [hunk 7]
```diff
@@ -104,6 +102,17 @@ class ResourceReservation(models.Model):
         help="Email of the user who created the reservation."
     )
 
+    repeat = fields.Selection([
+        ('once', 'Once'),
+        ('weakly', 'Every Week')], required=True,
```
Comment (MORPSmcaf) [inline resource_reservation/models/resource_reservation.py:107]: rename weakly to weekly

File: resource_reservation/models/resource_reservation.py [hunk 8]
```diff
@@ -142,22 +151,29 @@ def update_booking_status_confirm(self):
...
+            while_brake_counter = 0
+            while new_vals.get('end_datetime') <= repeat_date_str:
+                while_brake_counter += 1
+                start_date = fields.Datetime.from_string(new_vals['start_datetime'])
+                end_date = fields.Datetime.from_string(new_vals.get('end_datetime'))
+                start_date += timedelta(days=7)
+                end_date += timedelta(days=7)
+                new_vals['start_datetime'] = fields.Datetime.to_string(start_date)
+                new_vals['end_datetime'] = fields.Datetime.to_string(end_date)
+                if while_brake_counter > 50:
```
Comment (MORPSmcaf) [inline resource_reservation/models/resource_reservation.py:173]: Define 50 as constants at the start of code.

**Review by MORPSmcaf — APPROVED** [review]

**Pull request merged**